modules/http/src/test/java/org/mule/module/http/internal/request/grizzly/ProxyParamsTestCase.java

```diff
@@ -0,0 +1,160 @@
+ * LICENSE.txt file.
+ */
+
+package org.mule.module.http.internal.request.grizzly;
+
+import org.apache.commons.io.IOUtils;
+import org.glassfish.grizzly.http.server.HttpHandler;
+import org.glassfish.grizzly.http.server.HttpServer;
+import org.glassfish.grizzly.http.server.Request;
+import org.glassfish.grizzly.http.server.Response;
+import org.junit.*;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+import org.mule.api.MuleException;
+import org.mule.module.http.internal.ParameterMap;
+import org.mule.module.http.internal.domain.request.DefaultHttpRequest;
+import org.mule.module.http.internal.domain.request.HttpRequest;
+import org.mule.module.http.internal.request.DefaultProxyConfig;
+import org.mule.module.http.internal.request.HttpClientConfiguration;
+import org.mule.tck.junit4.AbstractMuleTestCase;
+import org.mule.tck.junit4.rule.DynamicPort;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Arrays;
+import java.util.Collection;
+
+import static org.eclipse.jetty.http.HttpMethod.GET;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.fail;
+
+@RunWith(Parameterized.class)
+public class ProxyParamsTestCase extends AbstractMuleTestCase
+{
+
+    public DynamicPort dynamicPort = new DynamicPort("port");
+    public DynamicPort proxyPort = new DynamicPort("proxyPort");
+
+    private static final String EXPECTED_RESULT = "result";
+    private static final String PROTOCOL = "http";
+    private static final String HOST = "localhost";
+    private static final String PATH = "retrieve";
+    private static final int TIMEOUT = -1;
+    private static final Boolean FOLLOW_REDIRECTS = Boolean.TRUE;
+    private MockServer server = new MockServer(dynamicPort.getNumber());
+    private GrizzlyHttpClient httpClient;
+
+    @Parameterized.Parameter()
+    public String nonProxyHosts;
+
+    @Parameterized.Parameters(name = "{0}")
+    public static Collection<Object[]> parameters()
+    {
+        return Arrays.asList(new Object[][] {
+                {""},
+                {"localhost"},
+                {"other|localhost"}
+        });
+    }
+    
+    @After
+    public void after() throws Exception
+    {
+        server.stop();
+        httpClient.stop();
+    }
+
+    @Test
+    public void testProxyNonProxyParams() throws Exception {
+        createProxyAndClient(nonProxyHosts);
+
+        final HttpRequest request = new DefaultHttpRequest(
+                PROTOCOL + "://" + HOST + ":" + dynamicPort.getNumber() + "/" + PATH, null, GET.asString(),
+                new ParameterMap(), new ParameterMap(), null);
+
+        if (nonProxyHosts != null && !nonProxyHosts.isEmpty()) {
+            // as we set the nonProxyHosts to localhost, the proxy will not block this request
+            InputStream responseStream = httpClient.sendAndReceiveInputStream(request, TIMEOUT, FOLLOW_REDIRECTS, null);
```

Comment:
This is not the primary method used in the client when used functionally. Though this test is technically correct it is not validating the real scenario where the feature will be used. Please refactor it to a functional one (XML based).